### ISSUE TITLE
`onPress` in `ToastProps`

### ISSUE THREAD

**Issue by cescoallegrini** [backpackapp-io/react-native-toast#36]
Hi! I built a custom wrapper for the toasts in my app using this lib, it's been pretty much straight forward, so thanks for your work.
Since I need to add listeners on the single toast instance, without using onToastPress on `Toasts`, I wanted to ask if this feature is on the development roadmap and what's the reason why you didn't include it on the single toast options.
Other than that, I guess that for now I'd need to manually add the `onPress` property in my wrapper and build a handler that executes that function in `onToastPress` using ids to match the right one.
The need of decentralized handlers comes from the lack of `onToastPress` to access component-specific methods or values such as _local_ react hooks, custom hooks or generic functions. Thanks in advance for your time

**Comment by nickdebaise** [comment]
You’re absolutely right that having decentralized handlers tied to the single toast instance would be super helpful. The main reason this isn’t included in the toast() API currently is due to potential issues that arise since toast() doesn’t depend on React or its lifecycle. If a component that calls toast() unmounts, it could lead to messy side effects, especially if the onPress function relies on local state, hooks, or context from the originating component.

To illustrate the issue, imagine this scenario:

```js
// Inside library's Toast Component
<Toast 
  onPress={(id) => {
    if(toast.onPress) toast.onPress(toast);
  }}
/>

// SomeComponent using toast()
useEffect(() => {
  toast("Hello!", {
    onPress: () => {
      console.log("Do something specific here");
    },
  });
}, []);
```

Now, if SomeComponent unmounts before the toast is pressed, the handlePress function is no longer part of the React tree, potentially leading to errors or unintended behavior.

Here are 2 solutions that I thought up:

1. Let developers manage side effects themselves: 

Implement the onPress behavior as-is and allow developers to handle side effects manually. This could involve ensuring that handlers don't rely on local component state or hooks that might be invalidated on unmount.

2. Expose an onToastPress listener in the useToaster() hook: 

This approach allows developers to handle toast presses centrally within a React component, ensuring access to the necessary React lifecycle and context. For example:
```js
const MyToastListener = () => {
  const { onToastPress } = useToaster();

  React.useEffect(() => {
    const unsubscribe = onToastPress((id) => {
      console.log(`Toast with ID ${id} pressed`);
      // Access hooks, state, or context here safely
    });

    return unsubscribe; // Clean up when the component unmounts
  }, [onToastPress]);

  return null; // This component only handles side effects
};
```
This second approach could make it easier to track IDs, manage state, and avoid lifecycle issues while still allowing flexibility.

I’d love to hear your thoughts on these solutions or if you have specific needs that might make one approach preferable. Thanks again for your thoughtful feedback—it’s incredibly valuable in improving the library!

**Comment by davidtran** [comment]
I prefer to have onPress as a param at toast.success() method. Let the dev manage the side effect by themselves.

**Comment by Tobbe** [comment]
I also want this feature now 🙂  And I'd also want something like `onToastHide()` but that also lets me know if the toast was hidden because it timed out, because the user swiped to close it or because it was programmatically dismissed.

**Comment by manhtruongwang** [comment]
Here how I use the accessibility to handle the onPress in root <Toast/> if any one interested
```
toast(message, {
   duration: 3000,
   accessabilityMessage: `/messages/${data.from_id}`,
});
```

And in my Toast, I config like this

```
<Toasts
      onToastPress={(toast) => {
      if (toast.accessabilityMessage) {
         router.push(toast.accessabilityMessage as any);
      }
   }}
/>
```

**Comment by nickdebaise** [comment]
Exciting news. The onPress toast prop was added alongside dismiss reasoning for onShow and onHide (also added to props list). You can see some examples here in the docs: [Examples in Docs](https://nickdebaise.github.io/packages/react-native-toast/api/toast-toast-handlers) , [onPress & handlers](https://nickdebaise.github.io/packages/react-native-toast/api/toast#onpress)

This will be included in the next release, or you can download the GitHub repo as the JS package for now. Please feel free to give your feedback, but closing for now.

**Comment by nickdebaise** [comment]
Released in 0.14.0
